tests/src/core/testqgslayerdefinition.cpp

```diff
@@ -90,6 +95,19 @@ void TestQgsLayerDefinition::testFindLayers()
   QCOMPARE( QgsProject::instance()->layerTreeRoot()->findLayers().at( 1 )->name(), QStringLiteral( "NewMemory" ) );
 }
 
+void TestQgsLayerDefinition::testLoadTopOfTree()
+{
+  QgsSettings settings;
+  settings.setEnumValue( QStringLiteral( "/qgis/layerTreeInsertionMethod" ), Qgis::LayerTreeInsertionMethod::TopOfTree );
```

Comment:
Can you add this kind of thing to the test init methods so that we don't mess with the real user profile settings when running this test? https://github.com/qgis/QGIS/blob/master/tests/src/core/testqgsapplication.cpp#L59